system/jlib/jptree.hpp

```diff
@@ -313,6 +313,22 @@ inline static bool isValidXPathChr(char c)
     return ('\0' != c && (isalnum(c) || strchr(validChrs, c)));
 }
 
+static const char *validXPathValueChrs = "!#$%^&()+,-.:;?@_{|}~"; //Do I miss any?
```

Comment:
Hmmm.  The level of validation could and probably should be a lot stricter.  For each request parameter we should first ask a few questions:
1. Is it ever stored in DALI/PTree as an element?  If so the allowed characters are a lot less.
2. Is it stored in Ptree as a string?
3. Can it be treated as a restricted name type... with a limited range of characters.
4. Is the parameter allowed to have wild cards?

For checking xpath elements rather than value strings there are existing methods in ptree.cpp
inline static bool isValidXPathStartChr(char c)
{
    return ('\0' != c && (isalpha(c) || strchr(validStartChrs, c)));
}

static const char *validChrs = ":_.-";
inline static bool isValidXPathChr(char c)
{
    return ('\0' != c && (isalnum(c) || strchr(validChrs, c)));
}

For a lot of parameters that are names... cluster names, etc.  We can be very strict.  Perhaps only allowing alphanum and '.' and '-' etc (what others?)?

Reply:
If it is stored as an element we might not even allow ':' which would act like a namespace... although ptree could still handle it so it may be ok depending on what it is.

Reply:
You could have at least 3 functions:

isValidXMLElementName(char *s);
isValidXPathValue(char *s);
isValidName(char *s, const char *extraAllowedCharacters);

example:
isValidName("my-secret.101", "-.");  //returns true
isValidName("my-secret.101[1]", "-.");  //returns false

Reply:
Good comments. I thought about to create this PR as a draft to collect comments.
I name it as validXPathValueChrs and, in this PR, it is only used to verify an xpath value: xpath='**value**'. I haven't thought about validating elements and names yet.

Reply:
All of the functions can have the default allowed characters and extraAllowedCharacters. The challenge is which characters should be in the default allowed.